### PR TITLE
Added memory-fork test for memory sanity test

### PR DESCRIPTION
**Pull request by harish-24**
Test has three scenarios each of which uses 85% of the free memory by creating processes and using some part of memory per process.

Signed-off-by: Harish <harish@linux.vnet.ibm.com>

### CODE REVIEW

File: memory/fork_mem.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    """
+
+    def setUp(self):
+        '''
+        Use 90% of memory with/without many process forked
+        '''
+
+        smm = SoftwareManager()
+        self.iter = self.params.get('iterations', default='10')
+        self.procs = self.params.get('procs', default='1')
```
Comment (narasimhan-v) [inline memory/fork_mem.py]: `int(self.params.get)`

File: memory/fork_mem.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        self.minmem = self.params.get('minmem', default='10')
```
Comment (narasimhan-v) [inline memory/fork_mem.py]: `int(self.params.get)`

File: memory/fork_mem.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+
+        if process.system('./forkoff %s %s %s' % (
+                          self.freemem / self.procs, self.procs, self.iter),
+                          shell=True, ignore_status=True) != 0:
+            self.fail("Test failed")
+
+        if process.system('./forkoff %s %s %s' % (
+                          self.minmem, self.freemem / self.minmem, self.iter),
+                          shell=True, ignore_status=True) != 0:
+            self.fail("Test failed")
```
Comment (narasimhan-v) [inline memory/fork_mem.py]: Same operation for 3 times, code can be optimized.

File: memory/fork_mem.py (hunk 1 above)
Comment (narasimhan-v) [reply memory/fork_mem.py]: Also check for non-zero value.

File: memory/fork_mem.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+        1. Total memory with one process
+        2. Split memeory between given processes
+        3. Maximum process with minimum memory (10 MB) per process
+        '''
+        os.chdir(self.teststmpdir)
+
+        if process.system('./forkoff %s %s %s'
+                          % (self.freemem, 1, self.iter),
+                          shell=True, ignore_status=True) != 0:
+            self.fail("Test failed")
```
Comment (narasimhan-v) [inline memory/fork_mem.py]: Would you want to fail fast or try all combinations and fail in the last ?

File: memory/fork_mem.py.data/forkoff.c [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+                        for (j=0; j<itterations; j++) {
+                                for (i = ptr; i < ptr + size - 1; i += psize) {
+					*i=(char)'i';
+                                }
+                        }
+			exit(0);
+		}
+	} while(--k);
+	while (procs-- && wait(&status));
+}
```
Comment (narasimhan-v) [inline memory/fork_mem.py.data/forkoff.c]: Suggestion: Can this go to LTP ?

File: memory/fork_mem.py (hunk 1 above)
Comment (harish-24) [reply memory/fork_mem.py]: Sure

File: memory/fork_mem.py (hunk 2 above)
Comment (harish-24) [reply memory/fork_mem.py]: Sure

File: memory/fork_mem.py (hunk 4 above)
Comment (harish-24) [reply memory/fork_mem.py]: All combinations would help, will change accordingly.

File: memory/fork_mem.py (hunk 3 above)
Comment (harish-24) [reply memory/fork_mem.py]: Sure, will change.

**Comment by harish-24** [comment]
@narasimhan-v Thanks for the review. Pushed the changes, please check.

File: memory/fork_mem.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        2. Split memeory between given processes
+        3. Maximum process with minimum memory (10 MB) per process
+        '''
+        os.chdir(self.teststmpdir)
+
+        self.run_test('./forkoff %s %s %s' % (self.freemem, 1, self.iter))
+        self.run_test('./forkoff %s %s %s' %
+                      (self.freemem / self.procs, self.procs, self.iter))
+        self.run_test('./forkoff %s %s %s' %
+                      (self.minmem, self.freemem / self.minmem, self.iter))
```
Comment (narasimhan-v) [inline memory/fork_mem.py]: Could still be optimised:
```
    def run_test(self, param1, param2, param3):
        cmd = "./forkoff %s %s %s" % (param1, param2, param3)
        if process.system(cmd, shell=True, ignore_status=True) != 0:
            self.fails.append(cmd)



        self.run_test(self.freemem, 1, self.iter)
        self.run_test(self.freemem / self.procs, self.procs, self.iter)
        self.run_test(self.minmem, self.freemem / self.minmem, self.iter)
```

**Comment by harish-24** [comment]
@narasimhan-v Thanks. Changed accordingly.

**Comment by harish-24** [comment]
@PraveenPenguin Addressed the suggested. Please have a look. Thanks.

**Pull request merged**